clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -3664,10 +3664,12 @@ static void GenerateMutualExclusionsChecks(const Record &Attr,
   if (Attr.isSubClassOf("TypeAttr"))
     return;
 
-  // This means the attribute is either a statement attribute or a decl
-  // attribute, find out which.
+  // This means the attribute is either a statement attribute, a decl
+  // attribute, or both; find out which.
   bool CurAttrIsStmtAttr =
       Attr.isSubClassOf("StmtAttr") || Attr.isSubClassOf("DeclOrStmtAttr");
+  bool CurAttrIsDeclAttr =
+      !CurAttrIsStmtAttr || Attr.isSubClassOf("DeclOrStmtAttr");
```

Comment:
@AaronBallman, isn't the second bool the same as:
!Attr.isSubClassOf("StmtAttr") || Attr.isSubClassOf("DeclOrStmtAttr");

Wouldn't this match anything that is not StmtAttr?

Reply:
That's correct.

There are three kinds of attributes (four-ish if you count pragmas, but those never get here): type, stmt, and decl. We bail out early if it's a type attribute on line 3664, so by this point it should either be a statement attr, a decl attr, or both.

Reply:
Thanks!

Reply:
How do you want to handle this PR?   Is there a separate PR to address statement attributes?   I just don't want us to lose track of it.  I will approve this one, assuming that one won't fall through the cracks.

Reply:
>> Is there a separate PR to address statement attributes?

I think so. Here is the discussion:
https://github.com/intel/llvm/pull/3507/#discussion_r608906356

Reply:
Statement attributes need more love; I'm working on that refactoring currently. If it looks like that refactoring has to be set down for some reason, I'll file an issue about it.

Reply:
@premanandrao, @AaronBallman, is this conversation resolved or I should wait for PR updates?

Reply:
I'm not expecting to change this PR further unless I've missed something or there are other concerns.

Reply:
> @premanandrao, @AaronBallman, is this conversation resolved or I should wait for PR updates?

Sorry, I should have been more explicit.   I considered this resolved.